crates/rpc/rpc-eth-api/src/helpers/fee.rs

```diff
@@ -142,6 +188,25 @@ pub trait EthFees: LoadFee {
                         rewards.push(block_rewards);
                     }
                 }
+
+                if let Some((pending_block, pending_receipts)) = pending_block {
+                    if fee_entries.len() < block_count as usize {
+                        let pending_entry = self.pending_block_to_fee_entry(&pending_block, &pending_receipts)?;
+                    
+                        base_fee_per_gas.push(pending_entry.base_fee_per_gas as u128);
+                        gas_used_ratio.push(pending_entry.gas_used_ratio);
+                        base_fee_per_blob_gas.push(pending_entry.base_fee_per_blob_gas.unwrap_or_default());
+                        blob_gas_used_ratio.push(pending_entry.blob_gas_used_ratio);
+    
+                        if let Some(percentiles) = &reward_percentiles {
+                            let mut block_rewards = Vec::with_capacity(percentiles.len());
+                            for &percentile in percentiles {
+                                block_rewards.push(self.approximate_percentile(&pending_entry, percentile));
+                            }
+                            rewards.push(block_rewards);
+                        }
+                    }
```

Comment:
could we instead push the pending fee entry to the list of entries in L177?

Reply:
ok